zbuf/order.go

```diff
@@ -20,3 +28,48 @@ func (o Order) String() string {
 	}
 	return "ascending"
 }
+
+func (o *Order) UnmarshalJSON(b []byte) error {
+	var s string
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+	switch s {
+	case "asc":
+		*o = false
+	case "desc":
+		*o = true
+	default:
+		return fmt.Errorf("bad serialization of zbuf.Order: %s", s)
+	}
+	return nil
+}
+
+func (o Order) MarshalJSON() ([]byte, error) {
+	s := "asc"
+	if o {
+		s = "desc"
+	}
+	return json.Marshal(s)
+}
+
+func (o Order) MarshalZNG(m *zson.MarshalZNGContext) (zng.Type, error) {
+	s := "asc"
+	if o {
+		s = "desc"
+	}
+	return m.MarshalValue(s)
+}
+
+func (o *Order) UnmarshalZNG(u *zson.UnmarshalZNGContext, zv zng.Value) error {
+	s := string(zv.Bytes)
+	switch s {
+	case "asc":
+		*o = false
+	case "desc":
+		*o = true
+	default:
+		return fmt.Errorf("bad zng serialization of zbuf.Order: %s", s)
+	}
+	return nil
+}
```

Comment:
Package `encoding/json` will use the `encoding.TextMarshaler` and `encoding.TextUnmarshaler` interfaces if a type implements them. The `zson` package should do likewise so we don't need two variants (JSON and ZNG) of these methods. Doesn't have to happen now, though.

Reply:
I agree. Issue #2550